test/unit/oceans/ui.test.js

```diff
@@ -197,3 +198,124 @@ describe('Words', () => {
     });
   });
 });
+
+describe('Train', () => {
+  let classifyFishStub;
+
+  beforeEach(() => {
+    classifyFishStub = sinon.stub(train, 'onClassifyFish');
+  });
+
+  afterEach(() => {
+    train.onClassifyFish.restore();
+  });
+
+  it('displays the current training count', () => {
+    setState({yesCount: 10, noCount: 23});
+    const wrapper = shallow(<Train {...DEFAULT_PROPS} />);
+    const trainCount = wrapper.find('#uitest-train-count');
+
+    expect(trainCount.text()).toEqual('33');
+  });
+
+  it('displays 999 if current training count is greater than 999', () => {
+    setState({yesCount: 1000, noCount: 1});
+    const wrapper = shallow(<Train {...DEFAULT_PROPS} />);
+    const trainCount = wrapper.find('#uitest-train-count');
+
+    expect(trainCount.text()).toEqual('999');
+  });
+
+  it('sets state to display confirmation dialog when erase icon is clicked', () => {
+    const initialState = getState();
+    expect(initialState.showConfirmationDialog).toBeFalsy();
+    expect(initialState.confirmationDialogOnYes).toBeNull();
+
+    const wrapper = shallow(<Train {...DEFAULT_PROPS} />);
+    const eraseIcon = wrapper.find('FontAwesomeIcon').at(0);
+    eraseIcon.simulate('click');
+
+    const newState = getState();
+    expect(newState.showConfirmationDialog).toBeTruthy();
+    expect(newState.confirmationDialogOnYes).not.toBeNull();
```

Comment:
Should we also test that the dialog has actually shown via the React DOM?

Reply:
the actual rendering happens in [`<UI/>`](https://github.com/code-dot-org/ml-activities/blob/e205ba00e4ff41ff37c7611fe041135f83c1ea22/src/oceans/ui.jsx#L1611-L1616), so i'll test that when i get there!